### PR TITLE
simple new spline limits

### PR DESCRIPTION
**Pull request by mo271**

### CODE REVIEW

**Comment by mo271** [comment]
Currently this does not work as intended, because there are image where during decoding in `QuantizedSpline::Decode` only one control point appears and this leaves the `manhattan_distance` zero, although the width might be very large.
Here is the image I'm testing with:
```
https://jxl-art.surma.technology/?zcode=rZJJTgMxEEXXrlP4Am3V5GmDBBKIFQJlwTqiG2JEOhFxNpye6g7iAuCSbJXf9y9Pz23sOy-McD-1t133WhBgc_xo8_R03s69fW17O8zX4_v51PfT3P0vB28tIvo_xerC_-Iy1D_bkISkjoNQLS6HqEhuwJAyZqc2SFGHQS2jUEuqC2RRrjUWEkMx5YJCGheF2CRp0uLQUBLJiFwN2Sot5ok5YXHkrLQJuLDlymXhZEXI6ovRIDVWUo1Mq5OkmCNj5Likl7g4ZBQ0VNdd2yGS2APlhdhuWdBcFjFQQl8ugwkAbufx51HhpvVxOtqvIIS7z-1-ejyc7G4G0wK0V__irzzB4B_WfjN1zxHhGw
```

File: lib/jxl/splines.cc [hunk 1]
```diff
@@ -467,12 +470,20 @@ Status QuantizedSpline::Dequantize(const Spline::Point& starting_point,
...
   }
+  uint64_t width_estimate = 0;
   for (int i = 0; i < 32; ++i) {
     const float inv_dct_factor = (i == 0) ? kSqrt0_5 : 1.0f;
     result.sigma_dct[i] =
         sigma_dct_[i] * inv_dct_factor * kChannelWeight[3] * inv_quant;
-  }
-
+    // If we include the factor kChannelWeight[3]=.3333f here, we get a
+    // realistic area esitmate. We leave it out to simplify the calculations,
```
Comment (veluca93) [inline lib/jxl/splines.cc:479]: typo

File: lib/jxl/splines.cc [hunk 2]
```diff
@@ -634,6 +645,19 @@ Status Splines::InitializeDrawCache(const size_t image_xsize,
       return JXL_FAILURE("Too many pixels covered with splines");
     }
   }
+  if (total_estimated_area_reached >
+      std::min((image_xsize * image_ysize + (uint64_t(1) << 18)),
+               (uint64_t(1) << 22))) {
+    JXL_WARNING(
```
Comment (veluca93) [inline lib/jxl/splines.cc:651]: TODO about making this a hard failure for L5 bitstreams?

**Review by veluca93 — DISMISSED** [review]

File: lib/jxl/splines.cc (hunk 2 above)
Comment (mo271) [reply lib/jxl/splines.cc:652]: done

**Review by veluca93 — DISMISSED** [review]

**Comment by mo271** [comment]
rebased

**Pull request merged**